### PR TITLE
add change name function

### PR DESCRIPTION
**Pull request by FrankSun96**
add change name in profile page

### CODE REVIEW

File: app/Http/Controllers/UpdatePersonalInformationController.php [hunk 1]
```diff
@@ -41,23 +41,27 @@ protected function update(Request $request)
...
+            'sex' => 'string',   
         ]);
 
+        $current_user_name = $request->name;
         $current_user_height = $request->height;
         $current_user_birthdate = $request->birthdate;
-        $current_user_sex = $request->sex;
+        $current_user_sex = is_null($request->sex) ? auth()->user()->sex : $request->sex;
         $current_user_id = auth()->user()->id;
+        // dd(auth()->user());
```
Comment (xingdawang) [inline app/Http/Controllers/UpdatePersonalInformationController.php]: remove this debug lane please.

File: app/Http/Controllers/UpdatePersonalInformationController.php [hunk 2]
```diff
@@ -41,23 +41,27 @@ protected function update(Request $request)
         // dd($request);
         // Check validation of the input method.
         $request->validate([
+            'name' => 'string',
             'height' => 'integer',
             'birthdate' => 'date',
-            'sex' => 'string',
+            'sex' => 'string',
```
Comment (xingdawang) [inline app/Http/Controllers/UpdatePersonalInformationController.php]: no need for empty string.

File: app/Http/Controllers/UpdatePersonalInformationController.php [hunk 3]
```diff
@@ -41,23 +41,27 @@ protected function update(Request $request)
...
         
         // Insert personal information into the database.
         DB::table('users')
             ->where('id', $current_user_id)
             ->update([
+                'name' => $current_user_name,
                 'height' => $current_user_height,
                 'birth_date' => $current_user_birthdate,
-                'sex' => $current_user_sex
+                'sex' => $current_user_sex,
```
Comment (xingdawang) [inline app/Http/Controllers/UpdatePersonalInformationController.php]: remove pending empty space.

File: app/Http/Controllers/UserHealthController.php [hunk 4]
```diff
@@ -42,6 +42,7 @@ public function valide_profile() {
         	'height_status' => isset($current_user->height),
         	'birthdate_status' => isset($current_user->birth_date),
         	'sex_status' => isset($current_user->sex),
+
```
Comment (xingdawang) [inline app/Http/Controllers/UserHealthController.php]: remove empty lane.

File: resources/views/users/profile.blade.php [hunk 5]
```diff
@@ -11,12 +11,27 @@
...
+                            </div>
+                        </div>
+
                         <!-- User height -->
                         <div class="form-group{{ $errors->has('height') ? ' has-error' : '' }}">
                             <label for="height" class="col-md-4 control-label">Height</label>
                             <div class="col-md-6">
                                 <div class="input-group">
-                                    <input id="height" type="text" class="form-control" name="height" value="{!! $current_user_height !!}" placeholder="e.g. 180" autofocus>
+                                    <input id="height" type="text" class="form-control" name="height" value="{!! $current_user_height !!}" placeholder="e.g. 180">
```
Comment (xingdawang) [inline resources/views/users/profile.blade.php]: why autofocus is removed?

**Review by xingdawang — COMMENTED** [review]
need more work.

File: resources/views/users/profile.blade.php (hunk 5 above)
Comment (FrankSun96) [reply resources/views/users/profile.blade.php]: name should be labeled with autofocus

**Pull request merged**